src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -640,7 +661,7 @@ bool dcd_edpt_open (uint8_t rhport, tusb_desc_endpoint_t const * desc_edpt)
   uint8_t const epnum = tu_edpt_number(desc_edpt->bEndpointAddress);
   uint8_t const dir   = tu_edpt_dir(desc_edpt->bEndpointAddress);
 
-  TU_ASSERT(epnum < ep_count);
+  TU_ASSERT(dcd_edpt_check_if_avaliable(rhport, dir));
```

Comment:
So, the assert will be: 

```suggestion
  TU_ASSERT(dcd_ep_available(rhport, dir));
```

Reply:
So, the idea here - to make an assert function name so clear to the user, so we don't need to add the additional LOG for this case. 
This relates to the https://github.com/espressif/esp-usb/pull/11#issuecomment-1948058363, so probably we can avoid changing the LOG level to show the problem to the user only with ASSERT + propper name for the function. 

IMO, the function could be next: 
1. `dcd_ep_available()`
2. `dcd_ep_occupied()`
3. `dcd_ep_alloc()`

I think that the option 2. is more preferable. 
@tore-espressif , @Dazza0 May I ask yours opinion about:
1. Changing the LOG level (from 0 to 1) to show the message about dcd unable to provide more EP, because driver run out of EPs. 
2. Make it only with ASSERT() and do not change the LOG level. In that case, either pick up the function name for the function which verify the EP availability and increase the number of used EPs, or offer any other name.

Thanks.

Reply:
@roma-jam @peter-marcisovsky For `dcd_edpt_open()`, I think it should return `false` rather than assert so that callers can handle the failure at runtime.

Currently, the callers of `usbd_edpt_open() -> dcd_edpt_open()` already call `TU_ASSERT` on failure.

Reply:
@roma-jam I left the log message, to warn user about the endpoints. Since I saw a [commit](https://github.com/espressif/esp-usb/pull/11/commits/3c51edf62b09ac2eb0c72aba17e3b68a5f84102c) in the esp_tinyusb v1.4.4 thus I suppose will we will be changinig it in the new release. Is that correct?

Reply:
Yep, that’s right. 
the default LOG level will be set to 1 after release.